### PR TITLE
bold class EN

### PR DESCRIPTION
**Pull request by jmealing**
Update the well-bold class component to be independent of a well to be more flexible in applying bold to any element, such as the Contact us band, or the About the institution component used in the ILP.

To do:
[ ] FR version

### CODE REVIEW

**Comment by Garneauma** [comment]
@jmealing I suggest adding the utility class `.fnt-bld` in WET-BOEW instead of renaming this component. We would then deprecate the `.well-bold` plugin.

You can simply add the class in "src/base/proximity/_base.scss", right under `.fnt-nrml`.

**Comment by jmealing** [comment]
@Garneauma I don't have that file anywhere `"src/base/proximity/_base.scss"` to be able to edit it

The only place in my folders I've downloaded from the repo where I see that class is the `theme-srv.css` and `theme.css` files. How do I get to the one you want me to edit?

File: components/bold/_base.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* Well bold provisional */
 
-.well.well-bold {
+.bold {
```
Comment (Garneauma) [inline components/bold/_base.scss:3]:
```suggestion
.bold-content, .well.well-bold {
```

File: components/bold/bold-doc-en.html [hunk 2]
```diff
@@ -2,11 +2,11 @@
 breadcrumbs: [
 	{ "title": "GCWeb home", "link": "https://wet-boew.github.io/GCWeb/home.html" }
 ]
-dateModified: "2022-04-28"
-description: "Documentation and working example for the well bold"
+dateModified: "2023-08-25"
+description: "Documentation and working example for bold class"
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:6]:
```suggestion
description: "Documentation and working example for bolded content"
```

File: components/bold/bold-doc-en.html [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 language: "en"
-altLangPage: "well-bold-doc-fr.html"
-title: "Well bold"
+altLangPage: "bold-doc-fr.html"
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:8]:
```suggestion
altLangPage: "bold-content-doc-fr.html"
```

File: components/bold/bold-doc-en.html [hunk 4]
```diff
@@ -2,11 +2,11 @@
...
+title: "bold"
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:9]:
```suggestion
title: "Bold content"
```

File: components/bold/bold-doc-en.html [hunk 5]
```diff
@@ -17,22 +17,22 @@
 	<dt>Type</dt>
 	<dd>Component</dd>
 	<dt>Last review</dt>
-	<dd>2022-04-28</dd>
+	<dd>2023-08-25</dd>
 	<dt>Conforming to</dt>
 	<dd>n/a</dd>
 	<dt>Technical name</dt>
-	<dd><code>well-bold</code></dd>
+	<dd><code>bold</code></dd>
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:24]:
```suggestion
	<dd><code>bold-content</code></dd>
```

File: components/bold/bold-doc-en.html [hunk 6]
```diff
@@ -17,22 +17,22 @@
...
 </dl>
 
 <h2>Purpose</h2>
-<p>This apply font weight bold to evey text element inside a well component and generates normal font weight when the element <code>strong</code> is applied.</p>
+<p>This apply font weight bold to evey text element inside a component and generates normal font weight when the element <code>strong</code> is applied.</p>
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:28]:
```suggestion
<p>This apply font weight bold to every text element inside the component and generates normal font weight when the element <code>strong</code> is applied.</p>
```

File: components/bold/bold-doc-en.html [hunk 7]
```diff
@@ -17,22 +17,22 @@
...
 
 <h2>Working example</h2>
 <ul>
-	<li><a href="well-bold-en.html">Working example</a></li>
+	<li><a href="bold-en.html">Working example</a></li>
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:32]:
```suggestion
	<li><a href="bold-content-en.html">Working example</a></li>
```

File: components/bold/bold-doc-en.html [hunk 8]
```diff
@@ -17,22 +17,22 @@
...
 </ul>
 
-<h3>Code</h3><pre><code>&lt;div class="well well-bold"&gt;
+<h3>Code</h3><pre><code>&lt;div class="well bold"&gt;
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:35]:
```suggestion
<h3>Code</h3><pre><code>&lt;div class="bold-content"&gt;
```

File: components/bold/bold-doc-en.html [hunk 9]
```diff
@@ -71,7 +71,7 @@ <h2>API (Component version 1.0)</h2>
 </table>
 
 <h3>CSS Class (v1.0)</h3>
-<p><code>well well-bold</code><p>
+<p><code>bold</code><p>
 
 <h3>Visual rendering (v1.0)</h3>
-<p>Font weight bold is applied to evey text element inside a well component and generates normal font weight when the element <code>strong</code> is applied.</p>
+<p>Font weight bold is applied to evey text element inside a component and generates normal font weight when the element <code>strong</code> is applied.</p>
```
Comment (Garneauma) [inline components/bold/bold-doc-en.html:77]:
```suggestion
<p>Font weight bold is applied to every text element inside the component and generates normal font weight when the element <code>strong</code> is applied.</p>
```

File: components/bold/bold-doc-fr.html
Comment (Garneauma) [inline components/bold/bold-doc-fr.html:1]: rename "bold-content-doc-fr.html"

File: components/bold/bold-en.html [hunk 10]
```diff
@@ -2,22 +2,22 @@
 breadcrumbs: [
 	{ "title": "GCWeb home", "link": "https://wet-boew.github.io/GCWeb/home.html" }
 ]
-dateModified: "2022-08-25"
-description: "Working example for the well bold"
+dateModified: "2023-08-25"
+description: "Working example for the bold class"
```
Comment (Garneauma) [inline components/bold/bold-en.html:6]:
```suggestion
description: "Working example for the bold content component."
```

File: components/bold/bold-en.html [hunk 11]
```diff
@@ -2,22 +2,22 @@
...
 language: "en"
-altLangPage: "well-bold-fr.html"
-title: "Well bold"
+altLangPage: "bold-fr.html"
+title: "Bold class"
```
Comment (Garneauma) [inline components/bold/bold-en.html:9]:
```suggestion
title: "Bold content"
```

File: components/bold/bold-en.html [hunk 12]
```diff
@@ -2,22 +2,22 @@
...
 ---
 
 <span class="wb-prettify all-pre hide"></span>
 
 <ul class="list-inline">
-	<li><a class="btn btn-primary" href="well-bold-doc-en.html">Documentation</a></li>
-	<li><a class="btn btn-primary" href="https://github.com/wet-boew/GCWeb/issues/new?title=WellBold">Questions or comments?</a></li>
+	<li><a class="btn btn-primary" href="bold-doc-en.html">Documentation</a></li>
```
Comment (Garneauma) [inline components/bold/bold-en.html:15]:
```suggestion
	<li><a class="btn btn-primary" href="bold-content-doc-en.html">Documentation</a></li>
```

File: components/bold/bold-en.html [hunk 13]
```diff
@@ -2,22 +2,22 @@
...
+	<li><a class="btn btn-primary" href="https://github.com/wet-boew/GCWeb/issues/new?title=Bold">Questions or comments?</a></li>
```
Comment (Garneauma) [inline components/bold/bold-en.html:16]:
```suggestion
	<li><a class="btn btn-primary" href="https://github.com/wet-boew/GCWeb/issues/new?title=Bold-content">Questions or comments?</a></li>
```

File: components/bold/index.json-ld [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+	"@context": {
+		"@version": 1.1,
+		"dct": "http://purl.org/dc/terms/",
+		"title": { "@id": "dct:title", "@container": "@language" },
+		"description": { "@id": "dct:description", "@container": "@language" },
+		"modified": "dct:modified"
+	},
+	"title": {
+		"en": "Bold",
+		"fr": "En gras"
```
Comment (Garneauma) [inline components/bold/index.json-ld:11]:
```suggestion
		"en": "Bold content",
		"fr": "Contenu en gras"
```

File: components/bold/index.json-ld [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+	},
+	"description": {
+		"en": "This apply font weight bold to evey text element inside a component and generates normal font weight when the element <code>strong</code> is applied.",
+		"fr": "X"
```
Comment (Garneauma) [inline components/bold/index.json-ld:15]:
```suggestion
		"en": "This component applies font weight bold to every text element inside it and generates normal font weight when the element <code>strong</code> is applied.",
		"fr": "Ce composant applique une épaisseur de police en gras à chaque élément de texte qu'il contient et génère une épaisseur de police normale lorsque l'élément <code>strong</code> est appliqué."
```

File: components/bold/index.json-ld [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+	},
+	"modified": "2023-08-25",
+	"componentName": "bold",
```
Comment (Garneauma) [inline components/bold/index.json-ld:18]:
```suggestion
	"componentName": "bold-content",
```

File: components/bold/index.json-ld [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+		"examples": [
+			{
+				"title": "Bold - working example",
+				"language": "en",
+				"path": "well-bold-en.html"
+			},
+			{
+				"title": "En gras - example pratique",
+				"language": "fr",
+				"path": "bold-fr.html"
```
Comment (Garneauma) [inline components/bold/index.json-ld:30]:
```suggestion
				"title": "Bold content - working example",
				"language": "en",
				"path": "bold-content-en.html"
			},
			{
				"title": "Contenu en gras - example pratique",
				"language": "fr",
				"path": "bold-content-fr.html"
```

File: components/bold/index.json-ld [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+		"docs": [
+		{
+			"title": "Bold",
+			"language": "en",
+			"path": "bold-doc-en.html"
+		},
+		{
+			"title": "En gras",
+			"language": "fr",
+			"path": "bold-doc-fr.html"
```
Comment (Garneauma) [inline components/bold/index.json-ld:42]:
```suggestion
			"title": "Bold content",
			"language": "en",
			"path": "bold-content-doc-en.html"
		},
		{
			"title": "Contenu en gras",
			"language": "fr",
			"path": "bold-content-doc-fr.html"
```

**Review by Garneauma — CHANGES_REQUESTED** [review]
Hi @jmealing, after discussing with @duboisp, we decided to keep the code in GCWeb. So, forget my last comment.

The new approach is to rename the class to `.bold-content` (but keep `.well.well-bold`) and rename the component to "Bold content".

We will deprecate `.well.well-bold`, but keep it for now.

Please add another working example above the `.well.well-bold` that does not contain a `.well` element. Keep the current example under it, but please add a message saying that it is now deprecated.

Also, make sure every file ends with a new empty line to pass build lint checker.

Thank you and let me know if you have any question!

**Comment by jmealing** [comment]
Thanks @Garneauma. I accepted your changes and will review the files in full and get it to pass the lint checker

**Review by Garneauma — CHANGES_REQUESTED** [review]
This is a partial review:

- Rename all pages to use "bold-content" instead of "bold". For example: `bold-content-doc-en.html`.
- Rename folder to "bold-content".
- Update `theme.scss` to use the correct file URL (line 190).
- Update documentation page with a working example of the new class and a guideline on how to use it.

**Comment by Garneauma** [comment]
@duboisp Something happened here. This PR went from 3 files changed to over 300. Would you know what caused that?

@jmealing

**Comment by jmealing** [comment]
@Garneauma I rebased, maybe something went wrong? @delisma

**Comment by jmealing** [comment]
@Garneauma @delisma @duboisp this one still has an issue when I rebased I ended up with a whole bunch of other files. Not sure why as I did the process to rebase that has been fine everywhere else.

**Comment by Garneauma** [comment]
@delisma I believe this PR also needs to be fixed.

**Comment by jmealing** [comment]
> @delisma I believe this PR also needs to be fixed.

We did go through it and fix it, what is still showing up as a problem for you?

**Comment by Garneauma** [comment]
There seems to be some messed up file references in the files changed tab:
- components/bold/_base.scss~refs/remotes/upstream/master
- components/bold/index.json-ld~refs/remotes/upstream/master

Also, you will need to fix the remaining build issues.

**Review by Garneauma — DISMISSED** [review]

**Comment by duboisp** [comment]
Pre-approved upon review and local re-testing.

**Review by duboisp — DISMISSED** [review]

**Review by duboisp — APPROVED** [review]
Reviewed,

We will need to change "deprecated" state by something saying this will be deprecated at the next major version.

**Review by Garneauma — APPROVED** [review]

**Pull request merged**